src/architect/project/project.utils.ts

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+import chalk from 'chalk';
+import execa from 'execa';
+import inquirer from 'inquirer';
+import PromptUtils from '../../common/utils/prompt-utils';
+import { Dictionary } from '../../dependency-manager/utils/dictionary';
+import inquirerPrompt from 'inquirer-autocomplete-prompt';
+import LocalPaths from '../../paths';
+
+interface Selection {
+  name: string,
+  type: string,
+  repository?: string,
+  'architect-file'?: string,
+}
+
+export default class ProjectUtils {
+  static async prompt(choices: Selection[], message: string): Promise<Selection> {
+    inquirer.registerPrompt('autocomplete', inquirerPrompt);
+    const answers: { selected: any } = await inquirer.prompt([
+      {
+        type: 'autocomplete',
+        name: 'selected',
+        message: message,
+        source: async (answers_so_far: any, input: string) => {
+          return choices.map((p) => ({ name: p.name, value: p }));
+        },
+      },
+    ]);
+    return answers.selected as Selection;
+  }
+
+  static async conversionPrompt(message: string): Promise<boolean> {
+    const answers: { selected: boolean } = await inquirer.prompt([
+      {
+        type: 'confirm',
+        name: 'selected',
+        message: message,
+      },
+    ]);
+    return answers.selected;
+  }
+
+  static async fetchJsonFromGitHub(url: string): Promise<Selection[]> {
+    const response = await axios.get(url)
+    .then((res: any) => res.data)
+    .catch((err: any) => {
+      throw new Error(`Failed to fetch ${url}`);
+    });
+    return response.choices as Selection[];
+  }
+
+  static async downloadGitHubRepos(selection: Selection, project_dir: string): Promise<void> {
+    // download repositories
+    if (selection.repository) {
+      await execa('git', ['clone', selection.repository.toString(), project_dir + '/' + selection.name.toLowerCase()], { stdio: 'ignore' });
+      await PromptUtils.oclifTimedSpinner(
+        `Pulling down GitHub Repository`,
+         selection.name.toLowerCase(),
+        `${chalk.green('✓')} ${selection.name.toLowerCase()}`,
+      );
+    }
+    await new Promise(resolve => setTimeout(resolve, 2000));
```

Comment:
Why do we wait for 2s after cloning?

Reply:
That is a great question - I have no clue; I'll remove it.

Reply:
Haha cool - was going to say if there's a reason we should add a comment since it looked random, but maybe was just random 😛. There's also a couple merge conflicts in the package.json/lock files but once fixed I think this is good to go

Reply:
Sounds good! got everything all resolved